sklearn_numba_dpex/kmeans/drivers.py

```diff
@@ -401,6 +404,80 @@ def _relocate_empty_clusters(
     )
 
 
+def prepare_data_for_lloyd(X_t, init, tol, copy_x):
```

Comment:
Can you document `prepare_data_for_lloyd` indicating that this is centering `X` for numerical stability?

Reply:
:heavy_check_mark: mostly reported [sklearn docstring](https://github.com/scikit-learn/scikit-learn/blob/17df37aee/sklearn/cluster/_kmeans.py#L332)